File: ui/libs/commons/src/lib/shared/models/rule-response.dto.ts
import { InventoryResponseDto } from './inventory-response.dto';
import { NotificationDto } from './notification.dto';

export interface RuleResponseDto {
  ruleCode?: string;
  _links?: any;
  results?: any;
  notifications?: NotificationDto[];
}

export interface PlasmaApheresisRuleResultDto {
  result: InventoryResponseDto[];
}

File: backend/shipping-service/src/main/java/com/arcone/biopro/distribution/shippingservice/application/dto/RuleResponseDTO.java
package com.arcone.biopro.distribution.shippingservice.application.dto;

import lombok.Builder;
import org.springframework.http.HttpStatus;

import java.io.Serializable;
import java.util.List;
import java.util.Map;

@Builder
public record RuleResponseDTO(
    HttpStatus ruleCode,
    Map<String, List<?>> results,
    List<NotificationDTO> notifications,
    Map<String, String> _links
) implements Serializable {
}
